src/ui/dialogs/customTuplets.ts

```diff
@@ -0,0 +1,94 @@
+// [Smoosic](https://github.com/AaronDavidNewman/Smoosic)
+// Copyright (c) Aaron David Newman 2021.
+import { SmoMeasure } from '../../smo/data/measure';
+import { SmoMeasureFormat, SmoMeasureFormatNumberAttributes, SmoMeasueFormatBooleanAttributes } from '../../smo/data/measureModifiers';
+import { SuiScoreViewOperations } from '../../render/sui/scoreViewOperations';
+import { DialogDefinition, SuiDialogParams } from './dialog';
+import { SuiComponentAdapter, SuiDialogAdapterBase } from './adapter';
+import { PromiseHelpers } from '../../common/promiseHelpers';
+import {MakeTupletOperation} from "../../smo/xform/operations";
+
+declare var $: any;
+
+/**
+ * @category SuiDialog
+ */
+export class SuiCustomTupletAdapter extends SuiComponentAdapter {
+    makeTuplet: MakeTupletOperation = {
+        numNotes: 3,
+        notesOccupied: 2,
+        ratioed: false,
+        bracketed: true
+    };
+    constructor(view: SuiScoreViewOperations) {
+        super(view);
+    }
+    async commit(){
+        await this.view.makeTuplet(this.makeTuplet);
+    }
+    async cancel() {
+    }
+    get numNotes() {
+        return this.makeTuplet.numNotes;
+    }
+    set numNotes(value: number) {
+        if (value < 1) {
```

Comment:
@AaronDavidNewman is there a way to do this validation in a better way using logic from components/dialog/etc..

Reply:
Not right now.  You could add a min and max element to the rocker, but then you either need to 1) create a custom derived Rocker component in the customTuplet.ts file, like we do with the button arrays in the note head dialogs, or 2) add optional min and max to SuiBaseComponentParams and add logic to rocker.ts to handle them.  I'm fine with either of these, there are pros-cons for each.

I've thought about moving away from the untyped component constructors toward more of a factory model, like we did with menus.  That would make it easier to customize components, and creating dialogs from external applications more straightforward.  But the current method is pretty streamlined so I'm still indecisive.

Reply:
One key use case is the 2 over 3 model for 6/8.  The current dialog lets you do this but there ends up being a dot on the notes:

![image](https://github.com/user-attachments/assets/a1c8543a-dcba-4b34-b70f-8e65516de739)

Do we need custom business logic just for 2/3 to get rid of that?  The dot and the tuplet tell you the same thing, I don't think it makes sense to have both.  I didn't look it in up Gould to see what is correct, but I don't remember ever seeing them together in music.

Reply:
@AaronDavidNewman Can you double-check? I think you've added a 2:2 tuplet instead of a 2:3 tuplet on a dotted eighth.

Reply:
ah, you're right.  It should be 2/3.  This looks great.